src/desktop/global_shortcuts.rs

```diff
@@ -0,0 +1,305 @@
+//! Register global shortcuts
+
+use std::{collections::HashMap, fmt::Debug, time::Duration};
+
+use futures_util::TryFutureExt;
+use serde::{Deserialize, Serialize};
+use zbus::zvariant::{
+    DeserializeDict, ObjectPath, OwnedObjectPath, OwnedValue, SerializeDict, Type,
+};
+
+use super::{HandleToken, Session, DESTINATION, PATH};
+use crate::{
+    helpers::{call_request_method, receive_signal, session_connection},
+    Error, WindowIdentifier,
+};
+
+#[derive(Clone, SerializeDict, Type, Debug, Default)]
+#[zvariant(signature = "dict")]
+struct NewShortcutInfo {
+    /// User-readable text describing what the shortcut does.
+    description: String,
+    /// The preferred shortcut trigger, defined as described by the "shortcuts" XDG specification. Optional.
+    preferred_trigger: Option<String>,
+}
+
+/// Shortcut descriptor used to bind new shortcuts in [`GlobalShortcuts::bind_shortcuts`]
+#[derive(Clone, Serialize, Type, Debug)]
+pub struct NewShortcut(String, NewShortcutInfo);
```

Comment:
Not sure if that name is a good choice, but I kinda don't have other ideas, perhaps ShortcutDescriptor or something like that would be better?

Reply:
What is actually the difference between NewShortcut & Shortcut? I only see that preferred_trigger is optional and it doesn't make much sense to have the function for creating one have an optional `preferred_trigger` and the other not being optional...

Reply:
When defining new shortcuts, you define, `description` and optionally `preferred_trigger` in XDG defined machine-readable format.

When you read shortcuts already defined previously, you get `description` and `trigger_description`, and both are only human-readable descriptions.

So the common denominator of those two types is `description`.